paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt

```diff
@@ -189,17 +189,20 @@ class Paparazzi @JvmOverloads constructor(
   fun <V : View> inflate(@LayoutRes layoutId: Int): V = layoutInflater.inflate(layoutId, null) as V
 
   fun snapshot(name: String? = null, composable: @Composable () -> Unit) {
-    val hostView = ComposeView(context)
-    // During onAttachedToWindow, AbstractComposeView will attempt to resolve its parent's
-    // CompositionContext, which requires first finding the "content view", then using that to
-    // find a root view with a ViewTreeLifecycleOwner
-    val parent = FrameLayout(context).apply { id = android.R.id.content }
-    parent.addView(hostView, LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT)
-    PaparazziComposeOwner.register(parent)
-    hostView.setContent(composable)
-    snapshot(parent, name)
-
-    forceReleaseComposeReferenceLeaks()
+    try {
+      val hostView = ComposeView(context)
+      // During onAttachedToWindow, AbstractComposeView will attempt to resolve its parent's
+      // CompositionContext, which requires first finding the "content view", then using that to
+      // find a root view with a ViewTreeLifecycleOwner
+      val parent = FrameLayout(context).apply { id = android.R.id.content }
+      parent.addView(hostView, LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT)
+      PaparazziComposeOwner.register(parent)
+      hostView.setContent(composable)
+      snapshot(parent, name)
```

Comment:
Compose references were not being released if `snapshot(parent, name)` throws an exception i.e. an assertion.

Reply:
let's scope the try block to surround only the snapshot call then.

Reply:
Makes sense 👍🏽 

Done